src/sync/secret-scanning-alerts.ts

```diff
@@ -0,0 +1,113 @@
+import { SecretScanningAlertResponseItem } from "../github/client/secret-scanning-alert.types";
+import { transformGitHubStateToJiraStatus } from "../transforms/transform-secret-scanning-alert";
+
+export const getSecretScanningAlertTask = async (
+	parentLogger: Logger,
+	gitHubClient: GitHubInstallationClient,
+	jiraHost: string,
+	repository: Repository,
+	cursor: string | undefined,
+	perPage: number,
+	messagePayload: BackfillMessagePayload) => {
+
+	const logger = parentLogger.child({ backfillTask: "Secret scanning alerts" });
+	const startTime = Date.now();
+
+	logger.info({ startTime }, "Secret scanning alerts task started");
+	const fromDate = messagePayload?.commitsFromDate ? new Date(messagePayload.commitsFromDate) : undefined;
+	const smartCursor = new PageSizeAwareCounterCursor(cursor).scale(perPage);
+
+	const { data: secretScanningAlerts  } = await gitHubClient.getSecretScanningAlerts(repository.owner.login, repository.name, {
+		per_page: smartCursor.perPage,
+		page: smartCursor.pageNo,
+		sort: "created",
+		direction: SortDirection.DES
+	});
+
+	if (!secretScanningAlerts?.length) {
+		logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: 0 }, "Backfill task complete");
+		return {
+			edges: [],
+			jiraPayload: undefined
+		};
+	}
+
+	if (areAllBuildsEarlierThanFromDate(secretScanningAlerts, fromDate)) {
+		logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: 0 }, "Backfill task complete");
+		return {
+			edges: [],
+			jiraPayload: undefined
+		};
+	}
+	logger.info(`Found ${secretScanningAlerts.length} secret scanning alerts`);
+	const nextPageCursorStr = smartCursor.copyWithPageNo(smartCursor.pageNo + 1).serialise();
+	const edgesWithCursor = [{ secretScanningAlerts, cursor: nextPageCursorStr }];
+
+	const jiraPayload = await transformSecretScanningAlert(secretScanningAlerts, repository, jiraHost, logger,  messagePayload.gitHubAppConfig?.gitHubAppId);
+
+	logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: jiraPayload?.vulnerabilities?.length }, "Backfill task complete");
+	return {
+		edges: edgesWithCursor,
+		jiraPayload
+	};
+};
+
+const areAllBuildsEarlierThanFromDate = (alerts: SecretScanningAlertResponseItem[], fromDate: Date | undefined): boolean => {
+
+	if (!fromDate) return false;
+
+	return alerts.every(alert => {
+		const createdAt = new Date(alert.created_at);
+		return createdAt.getTime() < fromDate.getTime();
+	});
+
+};
+
+
+const transformSecretScanningAlert = async (
+	alerts: SecretScanningAlertResponseItem[],
+	repository: Repository,
+	jiraHost: string,
+	logger: Logger,
+	gitHubAppId: number | undefined
+): Promise<JiraVulnerabilityBulkSubmitData> => {
+
+	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, jiraHost);
+
+	const vulnerabilities = alerts.map((alert) => {
+		return {
+			schemaVersion: "1.0",
+			id: `d-${transformRepositoryId(repository.id, gitHubClientConfig.baseUrl)}-${alert.number}`,
```

Comment:
Secret scanning vulnerability IDs should start with `s` to differentiate them from dependabot and code scanning